src/components/EventUserTable.js

```diff
@@ -27,25 +40,111 @@ export class EventUserTable extends Component {
+      .then(response => {
+        let heardFrom = {};
+        for (let i = 0; i < response.size; ++i) {
+          const temp = response.data[i];
+          if (temp.hasOwnProperty('heardFrom')) {
+            heardFrom[temp.heardFrom] = heardFrom.hasOwnProperty(temp.heardFrom) ? heardFrom[temp.heardFrom] + 1 : 1;
+          }
+        }
+        this.setState({ heardFrom })
+      })
+
+    const eventParams = new URLSearchParams({
       id: eventID
     });
 
-    fetchBackend('/events/getUsers?' + params, 'GET')
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
       .then(response => response.json())
-      .then(response => {
-        this.setState({ rows: response });
+      .then(async users => {
+        let obj = {
+          'registered': 0,
+          'checkedIn': 0,
+          'waitlisted': 0,
+          'cancelled': 0,
+          'faculties': {},
+          'years': {},
+          'dietary': {},
+          'genders': {},
+          'heardFrom': {}
+        };
+
+        users.forEach(user => {
+          if (user.hasOwnProperty('registrationStatus')) {
+            switch (user.registrationStatus) {
+              case REGISTRATION_STATUS.REGISTERED:
+                ++obj['registered'];
+                break;
+              case REGISTRATION_STATUS.CHECKED_IN:
+                ++obj['checkedIn'];
+                break;
+              case REGISTRATION_STATUS.WAITLISTED:
+                ++obj['waitlisted'];
+                break;
+              case REGISTRATION_STATUS.CANCELLED:
+                ++obj['cancelled'];
+                break;
+              default:
+                return;
+            }
+          }
+        })
+
+        obj['rows'] = users;
+        if (firstRender) {
+          users.forEach(user => {
+            if (user.faculty) {
+              obj.faculties[user.faculty] = obj.faculties[user.faculty] ? obj.faculties[user.faculty] + 1 : 1;
+            }
+            if (user.year) {
+              const yearInt = parseInt(user.year);
+              if (yearInt) {
+                obj.years[yearInt] = obj.years[yearInt] ? obj.years[yearInt] + 1 : 1;
+              }
+            }
+            if (user.diet) {
+              obj.dietary[user.diet] = obj.dietary[user.diet] ? obj.dietary[user.diet] + 1 : 1;
+            }
+            if (user.gender) {
+              obj.genders[user.gender] = obj.genders[user.gender] ? obj.genders[user.gender] + 1 : 1;
+            }
+          })
+
+        }
+        this.setState({
+          rows: obj.rows,
+          registered: obj.registered,
+          waitlisted: obj.waitlisted,
+          cancelled: obj.cancelled,
+          checkedIn: obj.checkedIn,
+          faculties: obj.faculties,
+          years: obj.years,
+          genders: obj.genders,
+          dietary: obj.dietary
+        });
       });
   }
 
   componentDidMount() {
-    this.getEventTableData(this.props.event.id);
+    this.getEventTableData(this.props.event.id, true);
```

Comment:
Let's make a different function to get the stats instead only doing half of `getEventTableData` on `firstRender`